api/lib/domain/services/authentication/oidc-authentication-service.js

```diff
@@ -191,103 +202,51 @@ class OidcAuthenticationService {
+  async getUserInfo({ idToken, accessToken }) {
+    let userInfo = jsonwebtoken.decode(idToken);
+    const missingRequiredClaims = this.#findMissingRequiredClaims(userInfo);
+    if (missingRequiredClaims.length > 0) {
+      userInfo = await this._getUserInfoFromEndpoint({ accessToken });
     }
 
-    return {
-      given_name: userInfoContent?.given_name,
-      family_name: userInfoContent?.family_name,
-      sub: userInfoContent?.sub,
-      nonce: userInfoContent?.nonce,
+    const pickedUserInfo = {
+      firstName: userInfo.given_name,
+      lastName: userInfo.family_name,
+      externalIdentityId: userInfo.sub,
+      nonce: userInfo.nonce,
     };
-  }
-
-  getUserInfoMissingFields({ userInfoContent }) {
-    const missingFields = [];
-    if (!userInfoContent.family_name) {
-      missingFields.push('family_name');
-    }
-    if (!userInfoContent.given_name) {
-      missingFields.push('given_name');
-    }
-    if (!userInfoContent.sub) {
-      missingFields.push('sub');
-    }
-
-    const thereIsAtLeastOneRequiredMissingField = missingFields.length > 0;
-    return thereIsAtLeastOneRequiredMissingField ? `Champs manquants : ${missingFields.join(',')}` : false;
-  }
-
-  async getUserInfo({ idToken, accessToken }) {
-    const { family_name, given_name, sub, nonce } = await jsonwebtoken.decode(idToken);
-    let userInfoContent;
-
-    const isMandatoryUserInfoMissing = !family_name || !given_name || !sub;
 
-    if (isMandatoryUserInfoMissing) {
-      userInfoContent = await this.getUserInfoFromEndpoint({ accessToken, userInfoUrl: this.userInfoUrl });
+    if (this.claimsToStore) {
+      this.claimsToStore.forEach((claim) => {
+        pickedUserInfo[claim] = userInfo[claim];
+      });
     }
 
-    return {
-      firstName: given_name || userInfoContent?.given_name,
-      lastName: family_name || userInfoContent?.family_name,
-      externalIdentityId: sub || userInfoContent?.sub,
-      nonce: nonce || userInfoContent?.nonce,
-    };
+    return pickedUserInfo;
   }
 
-  async createUserAccount({ user, externalIdentityId, userToCreateRepository, authenticationMethodRepository }) {
+  async createUserAccount({
+    user,
+    userInfo,
+    externalIdentityId,
+    userToCreateRepository,
+    authenticationMethodRepository,
+  }) {
     let createdUserId;
 
     await DomainTransaction.execute(async (domainTransaction) => {
       createdUserId = (await userToCreateRepository.create({ user, domainTransaction })).id;
 
+      let claimsToStoreWithValues;
+      if (this.claimsToStore) {
+        claimsToStoreWithValues = Object.fromEntries(
+          Object.entries(userInfo).filter(([key, _value]) => this.claimsToStore.includes(key)),
+        );
+      }
+
       const authenticationMethod = new AuthenticationMethod({
         identityProvider: this.identityProvider,
+        ...(claimsToStoreWithValues && { authenticationComplement: { ...claimsToStoreWithValues } }),
```

Comment:
```suggestion
      let claimsToStoreWithValues = {};
      if (this.claimsToStore) {
        claimsToStoreWithValues = Object.fromEntries(
          Object.entries(userInfo).filter(([key, _value]) => this.claimsToStore.includes(key)),
        );
      }

      const authenticationMethod = new AuthenticationMethod({
        identityProvider: this.identityProvider,
        ...(claimsToStoreWithValues && { authenticationComplement: { ...claimsToStoreWithValues } }),
```
Si la variable `claimsToStoreWithValues` n'est pas initialisée on risque d'avoir l'erreur suivante à la ligne 249. 

![Screenshot 2023-11-07 at 09 42 31](https://github.com/1024pix/pix/assets/6919604/b2963747-cfc6-4dc5-ac8f-87d7245850ed)